crates/commitlog/src/repo/fs.rs

```diff
@@ -0,0 +1,97 @@
+use std::io;
+use std::{
+    fs::{self, File},
+    path::PathBuf,
+};
+
+use log::debug;
+
+use super::Repo;
+
+const SEGMENT_FILE_EXT: &str = ".stdb.log";
+
+pub fn segment_file_name(offset: u64) -> String {
+    format!("{offset:0>20}{SEGMENT_FILE_EXT}")
```

Comment:
```suggestion
    // Left-pad the offset with zeros to 20 digits.
    format!("{offset:0>20}{SEGMENT_FILE_EXT}")
```

I can never remember the Rust format language, so anywhere we use a more-elaborate format specifier than `{}` or `{:?}`, I like a quick comment describing what it does.